src/Common/src/System/IO/PathInternal.Unix.cs

```diff
@@ -16,15 +16,7 @@ internal static partial class PathInternal
         internal static bool HasIllegalCharacters(string path, bool checkAdditional = false)
         {
             Debug.Assert(path != null);
-
-            foreach (char c in path)
-            {
-                // Same as InvalidPathChars, unrolled here for performance
-                if (c == InvalidPathChar)
-                    return true;
-            }
-
-            return false;
+            return path.IndexOf('\0') >= 0;
```

Comment:
It seemed like an unnecessary redirection so I hard-coded the value instead, but I can change it back to "InvalidPathChar" if that would be preferred.

Will the compiler inline the value?